src/dotnet/CommandLine/CommandLineApplication.cs

```diff
@@ -64,7 +64,21 @@ public CommandLineApplication Command(string name, bool throwOnUnexpectedArg = t
         public CommandLineApplication Command(string name, Action<CommandLineApplication> configuration,
             bool throwOnUnexpectedArg = true)
         {
-            var command = new CommandLineApplication(throwOnUnexpectedArg) { Name = name, Parent = this };
+            var command = new CommandLineApplication(throwOnUnexpectedArg) { Name = name };
+            return Command(command, configuration, throwOnUnexpectedArg);
+        }
+
+        public CommandLineApplication Command(CommandLineApplication command, bool throwOnUnexpectedArg = true)
+        {
+            return Command(command, _ => { }, throwOnUnexpectedArg);
+        }
+
+        public CommandLineApplication Command(
+            CommandLineApplication command,
+            Action<CommandLineApplication> configuration,
+            bool throwOnUnexpectedArg = true)
```

Comment:
I agree with you, but don't think it's necessary to add it here. It should be clear enough to the caller they shouldn't be passing in a null here.